### PR TITLE
server: support bugzilla parents and cherrypicks

### PR DESCRIPTION
**Pull request by AlexNPavel**
This PR adds support for handling bugzilla parents and cherrypicking bugzilla bugs to jira.

### CODE REVIEW

**Comment by AlexNPavel** [comment]
Currently waiting on 2 things before this can proceed:
1. Addition of the custom bugzilla dependency field to Jira.
2. A full list of mappings from Bugzilla Components+Subcomponents to Jira Components.

**Comment by AlexNPavel** [comment]
This PR has been updated based on the final decision of how BZ backports should be handled, adds new label copying functionality for ART, and adds the BZ Component/Subcomponent -> Jira Component mapping. We are still waiting on the actual `Blocked by Bugzilla` field to be added to OCPBUGS, so this is still using a placeholder field name and type.

**Review by bradmwilliams — APPROVED** [review]

**Review by bradmwilliams — APPROVED** [review]

**Pull request merged**
